### PR TITLE
Fix: Google Calendar integration

### PR DESCRIPTION
**Pull request by martiuh**
Now Google Calendar sends notifications whenever a CRUD event happens.

### CODE REVIEW

**Comment by martiuh** [comment]
Closes #10

File: server/models/bookingMethods/bookingBeforeSaveOperationHook.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    }
+  ]
+};
+
+const onlyInOfficeHours = (start, end, timezone = "America/Mexico_City") => {
+  const startDate = moment(start).tz(timezone);
+  const endDate = moment(end).tz(timezone);
+
+  // Check if is weekday
+  const isAvailableDay = day => officeHours.workingDays.includes(day);
```
Comment (mishelashala) [inline server/models/bookingMethods/bookingBeforeSaveOperationHook.js]: move functions outside `onlyInOfficeHours`

File: server/models/bookingMethods/bookingBeforeSaveOperationHook.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    isAvailableHour(startDate.format("HH:mm"), endDate.format("HH:mm"))
+  );
+};
+
+const cannotOverlap = roomName => ({
+  status: 422,
+  message: `There is already an appointment on ${roomName}`
+});
+
+const cannotScheduleForYesterday = () => ({
```
Comment (mishelashala) [inline server/models/bookingMethods/bookingBeforeSaveOperationHook.js]: move functions to `factories/ErrorFactory.js`

```js
const errorFactory = require('factories/ErroFactory');

errorFactory.cannotScheduleForYesterday();
```

File: server/models/bookingMethods/bookingBeforeSaveOperationHook.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    if (getActualDate() > start) {
+      return next(cannotScheduleForYesterday());
+    }
+
+    if (!onlyInOfficeHours(start, end)) {
+      return next(cannotScheduleOutsideOfficeHours());
+    }
+
+    Booking.find(
+      {
```
Comment (mishelashala) [inline server/models/bookingMethods/bookingBeforeSaveOperationHook.js]: create a query builder for this:

```js
// this is a provisional name, choose a better one
Booking.find(queryBuilder.noCollision({ roomId, start, end }));
```

**Review by mishelashala — CHANGES_REQUESTED** [review]

File: server/models/bookingMethods/bookingBeforeSaveOperationHook.js (hunk 1 above)
Comment (martiuh) [reply server/models/bookingMethods/bookingBeforeSaveOperationHook.js]: I convert them to `pure functions`

**Pull request merged**